### PR TITLE
Picker layout fix (WI:3391028) 

### PR DESCRIPTION
**Pull request by KAnder425**
There is a layout problem with Pickers that crops up in a limited set of scenarios.

The problem is the Pickers initially display with the combobox text sized as though the dropdown was expanded.  This only occurs with more than 2 items in a list, only when embedded within a ScrollView, and only when there is no height styling applied to the Picker. The effect is quite ugly since all subsequent layout is affected.

If the window is resized, or when the selection is changed, the layout finishes correctly. 
 
I have a fix for this problem that triggers a call to UpdateLayout (Xaml's) on the ComboBox only after new items have been added to the ComboBox. There is a deeper root cause to this issue that deserves further investigation, but this fix will unblock the use of Pickers embedded in Flyout's.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2584)

### CODE REVIEW

File: vnext/ReactUWP/Views/PickerViewManager.cpp [hunk 1]
```diff
@@ -158,6 +172,7 @@ void PickerShadowNode::RepopulateItems()
         comboboxItem.Foreground(BrushFrom(item["textColor"]));
       comboBoxItems.Append(comboboxItem);
     }
+    m_newItems = true;
```
Comment (randy-flynn) [inline vnext/ReactUWP/Views/PickerViewManager.cpp]: I expected this to be in updateProperties when the "items" property is set. Does it only matter on first creation or is it just that the scenarios you have are always populating before the first render? E.g. what happens if it starts with no items then two items are added? Maybe that already works right?

File: vnext/ReactUWP/Views/PickerViewManager.cpp [hunk 2]
```diff
@@ -35,13 +35,16 @@ class PickerShadowNode : public ShadowNodeBase
...
+  bool IsExternalLayoutDirty() const override { return m_newItems; }
+  void DoExtraLayoutPrep(YGNodeRef yogaNode) override;
 
 private:
   void RepopulateItems();
   static void OnSelectionChanged(IReactInstance& instance, int64_t tag, folly::dynamic&& value, int32_t selectedIndex, folly::dynamic&& text);
 
   folly::dynamic m_items;
   int32_t m_selectedIndex = -1;
+  bool m_newItems = false;
```
Comment (randy-flynn) [inline vnext/ReactUWP/Views/PickerViewManager.cpp]: I'd like to see m_newItems be something like m_hasNewItems so it doesn't sound like a collection of the items.

Comment (KAnder425) [reply vnext/ReactUWP/Views/PickerViewManager.cpp]: Yes, will do.

File: vnext/ReactUWP/Views/PickerViewManager.cpp (hunk 1 above)
Comment (KAnder425) [reply vnext/ReactUWP/Views/PickerViewManager.cpp]: The RepopulateItems method is called from updateProperties when the "items" property is set. Our code clears all the items out and then re-adds them all at once.

Comment (KAnder425) [reply vnext/ReactUWP/Views/PickerViewManager.cpp]: ..oh, to clarify, if there are no items initially the problem doesn't repro...then if the items are added later, the workaround kicks in.

Comment (randy-flynn) [reply vnext/ReactUWP/Views/PickerViewManager.cpp]: Ah, I see that now. Gotta do better with this web UI. I miss CodeFlow ;-). I thought this was in createView.

So, you are always setting it, regardless of count. Probably safest.

**Comment by KAnder425** [comment]
Yes, I used to have a check on item count, but I got rid of that for simplicity. I agree about CodeFlow vs. the WebUI. I’ve got a link to CodeFlow on GitHub for Microsoft reviewers. Do you not have that? Sent from Mail for Windows 10[18362 - 16.0.11802.31451] From: Randy FlynnSent: Thursday, June 6, 2019 4:36 PMTo: microsoft/react-native-windowsCc: Ken Anderson; AuthorSubject: Re: [microsoft/react-native-windows] Picker layout fix (WI:3391028) (#2584) @randy-flynn commented on this pull request.In vnext/ReactUWP/Views/PickerViewManager.cpp:> @@ -158,6 +172,7 @@ void PickerShadowNode::RepopulateItems()         comboboxItem.Foreground(BrushFrom(item["textColor"]));       comboBoxItems.Append(comboboxItem);     }+    m_newItems = true;Ah, I see that now. Gotta do better with this web UI. I miss CodeFlow ;-). I thought this was in createView.So, you are always setting it, regardless of count. Probably safest.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by KAnder425** [comment]
Oh, I also added a commit for the m_hasNewItems change. Sent from Mail for Windows 10[18362 - 16.0.11802.31451] From: kander425@gmail.comSent: Thursday, June 6, 2019 4:44 PMTo: microsoft/react-native-windows; microsoft/react-native-windowsCc: AuthorSubject: RE: Re: [microsoft/react-native-windows] Picker layout fix (WI:3391028) (#2584) Yes, I used to have a check on item count, but I got rid of that for simplicity. I agree about CodeFlow vs. the WebUI. I’ve got a link to CodeFlow on GitHub for Microsoft reviewers. Do you not have that? Sent from Mail for Windows 10[18362 - 16.0.11802.31451] From: Randy FlynnSent: Thursday, June 6, 2019 4:36 PMTo: microsoft/react-native-windowsCc: Ken Anderson; AuthorSubject: Re: [microsoft/react-native-windows] Picker layout fix (WI:3391028) (#2584) @randy-flynn commented on this pull request.In vnext/ReactUWP/Views/PickerViewManager.cpp:> @@ -158,6 +172,7 @@ void PickerShadowNode::RepopulateItems()         comboboxItem.Foreground(BrushFrom(item["textColor"]));       comboBoxItems.Append(comboboxItem);     }+    m_newItems = true;Ah, I see that now. Gotta do better with this web UI. I miss CodeFlow ;-). I thought this was in createView.So, you are always setting it, regardless of count. Probably safest.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Pull request merged**